### PR TITLE
Add dev-only Auth0 bypass flag (frontend half of no-auth path)

### PR DESCRIPTION
**Pull request by skearnes**
## Summary

Frontend half of the no-auth testing path (#664), the prerequisite for local dev and Playwright E2E without an Auth0 round-trip.

- New **DOM-free module** `common/noAuth.constants.ts` exposing `noAuth`, gated on a non-production build:
  ```ts
  isNoAuthEnabled(env) = !env.PROD && env.VITE_E2E_NO_AUTH === 'TRUE'
  ```
  so it can **never activate in a production bundle** regardless of the env var. The guard is a pure function and **unit-tested** (`noAuth.constants.test.ts`), including the security case (PROD ⇒ always false).
- When enabled, `useAuth` skips the Auth0 `loginWithRedirect`, injects a static dev token into the axios client, and provisions the mock user. Hook order is preserved (the effects branch internally rather than early-returning).
- Documented the flags in `.env.template`.

## Important: this is inert on its own
The **functional** path also needs the backend e2e override (`verify_access_token` short-circuit + `/auth/jit-provisioning` accepting the dev token) specced in #664. That backend half is security-sensitive and only meaningfully verifiable with the full stack, so it will land together with the Playwright E2E job (#665) that exercises it end-to-end. Merging this FE flag alone changes nothing in production (guarded) and nothing in normal dev (flag off by default).

## Test plan
- [x] `npm run test` (guard unit tests pass, incl. PROD security guard)
- [x] `tsc`, `eslint`, `prettier --check`, `npm run build` clean

Part of #662. Refs #664.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h3>Greptile Summary</h3>

This PR introduces the frontend half of a dev/test Auth0 bypass, allowing local development and Playwright E2E tests to skip the Auth0 round-trip when `VITE_E2E_NO_AUTH=TRUE` is set in a non-production build.

- A new `noAuth.constants.ts` module exports a build-time `noAuth` flag gated on `!PROD && VITE_E2E_NO_AUTH === 'TRUE'`, with a unit-tested pure-function form; `e2eDevToken` is conditionally set to `''` in all non-noAuth builds so the fallback literal is absent from production bundles.
- `useAuth.ts` gains two new early-exit guards (login redirect, user provisioning) plus a dedicated `[dispatch]`-only effect for the `noAuth` user-creation path, cleanly separating it from the Auth0-dependent dep array that drove the spurious re-dispatch concern raised in a previous review.

<h3>Confidence Score: 5/5</h3>

Safe to merge — the bypass is build-time gated on !PROD and changes nothing in production or in normal dev without the flag explicitly set.

All noAuth code paths are unreachable in production bundles because noAuth resolves to a build-time constant false whenever import.meta.env.PROD is true. The previous concerns around spurious re-dispatches and the stray fallback string in production bundles are both addressed in this revision. The frontend change is inert on its own until the backend e2e mode lands.

No files require special attention.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| ui/src/common/noAuth.constants.ts | Introduces the noAuth flag and e2eDevToken; both gated on the PROD guard so production bundles carry no live auth-bypass code or stray string literals. |
| ui/src/common/noAuth.constants.test.ts | Unit tests cover the enabled case, the PROD security guard (always false), and several flag-value edge cases (absent, lowercase 'true', 'false'). |
| ui/src/common/hooks/useAuth.ts | Adds four noAuth guards across existing effects and a new [dispatch]-only effect for mock-user provisioning; hook call order is preserved and the re-dispatch concern from the previous review is addressed. |
| ui/.env.template | Documents the two new opt-in env vars with commented-out defaults; no functional change. |

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant C as Component
    participant uA as useAuth
    participant nA as noAuth.constants
    participant ax as axiosInstance
    participant A0 as Auth0
    participant BE as Backend

    Note over nA: Build-time: noAuth = !PROD && VITE_E2E_NO_AUTH==='TRUE'

    alt "noAuth === false (normal path)"
        C->>uA: mount
        uA->>A0: loginWithRedirect (if not authenticated)
        A0-->>uA: "isAuthenticated = true"
        uA->>ax: setAccessTokenGetter(getAccessTokenSilently)
        uA->>A0: getIdTokenClaims() + getAccessTokenSilently()
        A0-->>uA: idToken, accessToken
        uA->>BE: POST /auth/jit-provisioning (real tokens)
        BE-->>uA: user provisioned
    else "noAuth === true (dev/E2E bypass)"
        C->>uA: mount
        Note over uA: Effect 1: skip loginWithRedirect
        uA->>ax: "setAccessTokenGetter(() => e2eDevToken)"
        uA->>BE: "POST /auth/jit-provisioning (e2eDevToken, requires #664 backend)"
        BE-->>uA: mock user provisioned
    end
```
</details>

<sub>Reviews (2): Last reviewed commit: ["Address Greptile review on no-auth flag"](https://github.com/open-reaction-database/ord-app/commit/4b3c029fcd410a996ad72153f0774a40f6e77eb7) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=34737313)</sub>

### CODE REVIEW

File: ui/src/common/hooks/useAuth.ts [hunk 1]
```diff
@@ -40,12 +46,24 @@ export function useAuth() {
...
       setAccessTokenGetter(getAccessTokenSilently);
     }
   }, [isAuthenticated, getAccessTokenSilently]);
 
   useEffect(() => {
+    if (noAuth) {
+      // Provision the mock user with the static dev token (requires the backend e2e mode, #664).
+      dispatch(createUser({ access_token: e2eDevToken, id_token: e2eDevToken }));
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (skearnes) [reply ui/src/common/hooks/useAuth.ts:72]: Done in 4b3c029 — moved the mock-user provisioning into its own `[dispatch]`-only effect so Auth0 reference changes can't re-fire it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: ui/src/common/noAuth.constants.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * Hard-gated on a non-production build (`!PROD`) so it can NEVER activate in a production
+ * bundle, regardless of the env var. Exported as a pure function so the guard is unit-testable.
+ * Kept in its own (DOM-free) module so the guard can be tested without a browser environment.
+ */
+export const isNoAuthEnabled = (env: { PROD: boolean; VITE_E2E_NO_AUTH?: string }): boolean =>
+  !env.PROD && env.VITE_E2E_NO_AUTH === 'TRUE';
+
+export const noAuth = isNoAuthEnabled(import.meta.env);
+
+export const e2eDevToken = (import.meta.env.VITE_E2E_DEV_TOKEN as string) || 'e2e-dev-token';
```
[the remark this answers is not in the document]
Comment (skearnes) [reply ui/src/common/noAuth.constants.ts]: Done in 4b3c029 — applied your suggestion; `e2eDevToken` is now gated on `noAuth` so prod gets an empty string.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
